### PR TITLE
#170 nav tweaks

### PR DESCRIPTION
**Pull request by gitcow**
- Fixed proposal show page nav highlight issue.
- Using path helpers now.
- Added support for nil dependencies.
- Eliminated redundancy in path mappings.

### CODE REVIEW

**Comment by mghaught** [comment]
👍  Works well.

File: app/controllers/concerns/activate_navigation.rb [hunk 1]
```diff
@@ -7,43 +7,79 @@ module ActivateNavigation
...
+
+  def matches_nav_path?(key, paths={})
+    if paths.is_a?(String)
+      return paths==request.path
+
+    elsif paths.is_a?(Hash)
+      return matches_nav_path?(nil, paths.values) if key.nil?
+      return paths.any?{|k, p| k==key && matches_nav_path?(nil, p) } # only recurse if the key matches
+
+    elsif paths.is_a?(Array) || key.nil?
```
Comment (gitcow) [inline app/controllers/concerns/activate_navigation.rb]: oops need to get rid of that || key.nil? bit. Leftover from an earlier version

File: app/controllers/concerns/activate_navigation.rb [hunk 2]
```diff
@@ -7,43 +7,79 @@ module ActivateNavigation
...
+            add_path(:event_staff_program_proposals, current_event),
+            add_path(:event_staff_program_proposal, current_event, @proposal)
+        ],
+        'event-program-sessions-link' => add_path(:event_staff_program_sessions, current_event),
+        'event-program-speakers-link' => add_path(:event_staff_program_speakers, current_event),
+    }
+  end
+
+  def add_path(sym, *deps)
+    send(sym.to_s + '_path', *deps) unless deps.include?(nil) # don't generate the path unless all dependencies are present
```
Comment (gitcow) [inline app/controllers/concerns/activate_navigation.rb:85]: Couple things I don't like about this:
1) this is a controller concern, so this should definitely be declared private.
2) converting the sym to a string uses a bit of memory each time. Why not just pass in the whole name of the helper as a string? Readability with the symbols is nice though, and if the method can't be made private, it's definitely safer than having a random string + dependencies passed in. Confirm that a concern's private methods keep their privateness when they're included

Comment (mghaught) [reply app/controllers/concerns/activate_navigation.rb:85]: Sure, I can see all that.  

There is another drawback with this approach.  It requires you to match routes exactly.  

What seems like a more elegant solution is to use pattern matching on routes.  Thus you could use simplified regexes to wildcard sections of the app by url path.  

What you've got here is good for now and we can look at simplifying how to match a nav item's key with page urls another time.

**Pull request merged**